nuvla/job_engine/connector/kubernetes.py

```diff
@@ -132,6 +141,109 @@ def log(self, component: str, since: datetime, lines: int,
         return self.k8s.log(component, since, lines, namespace)
 
 
+class HelmAppMgmt(Connector, ABC):
+    """Class providing Helm application management functionalities using.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.helm = Helm(_NUVLAEDGE_SHARED_PATH, **kwargs)
+
+    @staticmethod
+    def helm_release_name(string: str):
+        return 'release-' + string
+
+    def get_helm_release(self, helm_release, namespace) -> dict:
+        release = {}
+        releases = self.helm.list(namespace, release=release)
+        if releases:
+            return releases[0]
+        log.warning(f'Helm release "{helm_release}" not found in '
+                    f'namespace {namespace}.')
+        return release
+
+    def _op_install_upgrade(self, op: str, deployment: dict, **kwargs) -> \
+            [str, List[dict], dict]:
+        namespace = kwargs['name']
+        helm_release = self.helm_release_name(namespace)
+
+        helm_repo_cred = kwargs.get('helm_repo_cred')
+
+        app_content = Deployment.module_content(deployment)
+        helm_repo_url = app_content.get('helm-repo-url')
+        chart_name = app_content.get('helm-chart-name')
+        version = app_content.get('helm-chart-version')
+        helm_absolute_url = app_content.get('helm-absolute-url')
+        chart_values_yaml = app_content.get('helm-chart-values')
+
+        env = kwargs.get('env')
+
+        interpolate_and_store_files(env, kwargs.get('files'))
+
+        if chart_values_yaml and env:
+            chart_values_yaml = string_interpolate_env_vars(chart_values_yaml, env)
+
+        registries_auth = kwargs.get('registries_auth')
+        # FIXME: check how to handle the secrets during upgrade.
```

Comment:
FIXME note is it for future work ?

Reply:
Yes. I created a ticket to handle this: https://github.com/SixSq/tasklist/issues/3150

Between the app upgrades, the creds of the registries can change. Our strategy can be as simple as deleting the current creds and creating new ones.